### PR TITLE
VTCCA-5792 added optional listYears to model & included LIST_YEAR action

### PR DESCRIPTION
**Pull request by zachhall1234**

### CODE REVIEW

**Comment by zachhall1234** [comment]
Resolving PR bot comments, will push up shortly.

File: app/models/propertyrepresentation/AgentAppointmentChangeRequest.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  val submitAgentAppointmentRequest: Form[AgentAppointmentChangeRequest] =
+    Form(
+      mapping(
+        "agentCode"       -> longNumber,
+        "action"          -> text.verifying(s => Try(AppointmentAction.withName(s)).toOption.isDefined),
+        "scope"           -> text.verifying(s => Try(AppointmentScope.withName(s)).toOption.isDefined),
+        "propertyLinkIds" -> optional(list(text)),
+        "listYears"       -> optional(list(text))
+      )(AgentAppointmentChangeRequest.apply)(AgentAppointmentChangeRequest.unapply)
+    )
```
[the remark this answers is not in the document]
Comment (zachhall1234) [reply app/models/propertyrepresentation/AgentAppointmentChangeRequest.scala]: Yep that is a good point, i've updated PR to remove the form mapping.

File: it/connectors/propertyLinking/PropertyLinkConnectorISpec.scala [hunk 2]
```diff
@@ -60,4 +60,33 @@ class PropertyLinkConnectorISpec extends ISpecBase {
...
+      val jsonResponse = Json.obj("appointmentChangeId" -> "1")
+
+      stubFor(
+        post("/property-linking/my-organisation/agent/submit-appointment-changes")
+          .willReturn(
+            aResponse
+              .withStatus(CREATED)
+              .withBody(jsonResponse.toString())
+          )
+      )
```
Comment (kabuumu) [inline it/connectors/propertyLinking/PropertyLinkConnectorISpec.scala:87]: Can we add a check on the stub to verify that the correct body has been passed please? There is an example of how to do this at https://github.com/hmrc/business-rates-check/blob/main/it/uk/gov/voa/businessrates/check/controllers/CheckBuilderControllerISpec.scala#L121

Comment (zachhall1234) [reply it/connectors/propertyLinking/PropertyLinkConnectorISpec.scala:87]: Yep no worries, added this request body check in last commit.

**Review by kabuumu — APPROVED** [review]

**Pull request merged**